### PR TITLE
Add tests for lru-cache.hh

### PR DESCRIPTION
**Pull request by gilligan**
Simple set of tests for a simple lru cache data structure ;)

### CODE REVIEW

File: src/libutil/tests/lru-cache.cc [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    TEST(LRUCache, eraseFromEmptyCache) {
+        LRUCache<std::string, std::string> c(10);
+        c.erase("foo");
+        ASSERT_EQ(c.size(), 0);
+    }
+
+    TEST(LRUCache, eraseMissingFromNonEmptyCache) {
+        LRUCache<std::string, std::string> c(10);
+        c.upsert("one", "eins");
+        c.erase("foo");
```
Comment (jbit) [inline src/libutil/tests/lru-cache.cc:118]:
```suggestion
        ASSERT_EQ(c.erase("foo"), false);
```

File: src/libutil/tests/lru-cache.cc [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        ASSERT_EQ(c.size(), 1);
+        ASSERT_EQ(c.get("one").value_or("error"), "eins");
+    }
+
+    TEST(LRUCache, eraseFromNonEmptyCache) {
+        LRUCache<std::string, std::string> c(10);
+        c.upsert("one", "eins");
+        c.erase("one");
```
Comment (jbit) [inline src/libutil/tests/lru-cache.cc:126]:
```suggestion
        ASSERT_EQ(c.erase("one"), true);
```

File: src/libutil/tests/lru-cache.cc [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        ASSERT_EQ(c.size(), 0);
+    }
+
+    /* ----------------------------------------------------------------------------
+     * erase
+     * --------------------------------------------------------------------------*/
+
+    TEST(LRUCache, eraseFromEmptyCache) {
+        LRUCache<std::string, std::string> c(10);
+        c.erase("foo");
```
Comment (jbit) [inline src/libutil/tests/lru-cache.cc:111]: You might also want to check the return value of `erase`:
```suggestion
        ASSERT_EQ(c.erase("foo"), false);
```

**Comment by gilligan** [comment]
Thanks for the review @jbit and thanks for fixing & merging @edolstra - i didn’t have the time to address those right away

**Pull request merged**